labs/SAW/SAW.md

```diff
@@ -41,30 +41,106 @@ tools. As well, there are some examples
+- Output(s)
+- Precondition(s)
+- Postcondition(s)
+
+This information is packaged into a script for SAW to process. This
+script can either be written in Python or SAW's special purpose
+language, SAWScript. For this lab, the scripts are written in Python
+since, as a general purpose programming language, it provides a wider
+range of techniques when expressing properties and it is much more
+widely known than SAWScript. Python also fairly easily integrates into
+CI systems.
+
+Combining all of these different languages (Python, C, LLVM,
+SAWScript, Cryptol) and tools together into one workflow is definitely
+a challenging process, especially for newcomers. However, learning
+how to use these components provides a very powerful method for
+developing cryptographic implementations and proving them correct and
+secure.
+
+Before setting up the lab, let's recap the overall SAW workflow using the
+diagram below.
+
+![SAW Workflow Overview](../../misc/SAWWorkflowOverview.png)
+
+1. A developer begins with a math paper algorithm specification that
+defines how an algorithm behaves.
+2. They then take that paper and write up a version of it in Cryptol
+(or grab one that already exists). Trust is gained in the Cryptol
+specification via properties defined in the paper and proven in the
+Cryptol specification.
+3. The developer beings implementing the algorithm in a general
+   purpose programming language such as C/C++ or Java.
+4. As they develop their implementation, SAW is used to verify that
+functions/methods behave according to the properties in a Cryptol
+specification.
+5. The implementation is continuously improved and tested until
+   verified by SAW.
+
 # Setting Everything Up
 
-To run any of the examples in this lab, you need to first start the
-Software Analysis Workbench (SAW) remote API (`saw-remote-api`).  If
-you are using the development container that comes with this course
-(`ghcr.io/weaversa/cryptol-course`), you can enter the following
-command in your terminal:
+To run any of the examples in this lab, first start the Software
+Analysis Workbench (SAW) remote API (`saw-remote-api`). If using the
+development container that comes with this course
+(`ghcr.io/weaversa/cryptol-course`), enter the following command in
+a terminal:
 
 ```sh
 $ start-saw-remote-api
 ```
 
 Otherwise, this tool can be installed by following the instructions
 for SAW found in the [Installation lab](../../INSTALL.md). Once
-installed, to run `saw-remote-api`, enter the following commands into
-your terminal:
+installed, to run `saw-remote-api`, enter the following commands in
+a terminal:
 
 ```sh
 $ export SAW_SERVER_URL=http://0.0.0.0:36691
 $ saw-remote-api http --host 0.0.0.0 --port 36691 / &
 ```
 
-Congrats! You now have a server version of SAW running in the
-background ready to accept commands on port `36691`.
+Congrats! A server version of SAW is now running in the background
+ready to accept commands on port `36691`.
 
 # SAW and Python

```

Comment:
camelCase [class names](https://peps.python.org/pep-0008/#class-names)?

Reply:
Key words from the style guide being "should normally". I suppose we have an abnormal situation here :)